File: 504.Base-7.go
// https://leetcode.com/problems/base-7/description/
//
// algorithms
// Easy (43.96%)
// Total Accepted:    34.2k
// Total Submissions: 77.8k
// beats 100.0% of golang submissions

package leetcode

import "strconv"

func convertToBase7(num int) string {
	if num == 0 {
		return "0"
	}
	symbol := ""
	if num < 0 {
		symbol = "-"
		num = -num
	}

	return symbol + strconv.FormatInt(int64(num), 7)
}

File: 504.Base-7.java
// https://leetcode.com/problems/base-7/
//
// algorithms
// Easy (45.38%)
// Total Accepted: 47,025
// Total Submissions: 103,620


class Solution {
    public String convertToBase7(int num) {
        int flag = 1;
        if (num < 0) {
            flag = -1;
            num = -num;
        }

        int base = 1;
        while (base * 7 <= num) {
            base *= 7;
        }

        StringBuilder sb = new StringBuilder();
        while (num > 0 || base > 0) {
            sb.append(num / base);
            num %= base;
            base /= 7;
        }

        if (flag == -1) {
            sb.insert(0, '-');
        }

        return sb.toString();
    }
}